components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretTypeManagementService.java

```diff
@@ -0,0 +1,256 @@
+        return secretTypeAdd;
+    }
+
+    private APIError handleSecretMgtException(SecretManagementException e, SecretManagementConstants.ErrorMessage
+            errorEnum, String data) {
+
+        ErrorResponse errorResponse = getErrorBuilder(errorEnum, data).build(log, e, errorEnum.getDescription());
+        Response.Status status;
+        if (e instanceof SecretManagementClientException) {
+            if (e.getErrorCode() != null) {
+                String errorCode = e.getErrorCode();
+                errorCode = errorCode.contains(CONFIG_MGT_ERROR_CODE_DELIMITER) ? errorCode :
+                        ERROR_PREFIX + errorCode;
+                errorResponse.setCode(errorCode);
+            }
+            errorResponse.setDescription(e.getMessage());
+            status = Response.Status.BAD_REQUEST;
+        } else if (e instanceof SecretManagementServerException) {
+            if (e.getErrorCode() != null) {
+                String errorCode = e.getErrorCode();
+                errorCode = errorCode.contains(CONFIG_MGT_ERROR_CODE_DELIMITER) ? errorCode :
+                        ERROR_PREFIX + errorCode;
+                errorResponse.setCode(errorCode);
+            }
+            errorResponse.setDescription(e.getMessage());
+            status = Response.Status.INTERNAL_SERVER_ERROR;
+        } else {
+            status = Response.Status.INTERNAL_SERVER_ERROR;
+        }
+        return new APIError(status, errorResponse);
+    }
+
+    /**
+     * Handle exceptions generated in API.
+     *
+     * @param status HTTP Status.
+     * @param error  Error Message information.
+     * @return APIError.
+     */
+    private APIError handleException(Response.Status status, SecretManagementConstants.ErrorMessage error,
+                                     String data) {
+
+        return new APIError(status, getErrorBuilder(error, data).build());
+    }
+
+    /**
+     * Return error builder.
+     *
+     * @param errorMsg Error Message information.
+     * @return ErrorResponse.Builder.
+     */
+    private ErrorResponse.Builder getErrorBuilder(SecretManagementConstants.ErrorMessage errorMsg,
+                                                  String data) {
+
+        return new ErrorResponse.Builder().withCode(errorMsg.getCode()).withMessage(errorMsg.getMessage())
+                .withDescription(includeData(errorMsg, data));
+    }
+
+    /**
+     * Include context data to error message.
+     *
+     * @param error Error message.
+     * @param data  Context data.
+     * @return Formatted error message.
+     */
+    private static String includeData(SecretManagementConstants.ErrorMessage error, String data) {
+
+        String message;
+        if (StringUtils.isNotBlank(data)) {
+            message = String.format(error.getDescription(), data);
+        } else {
+            message = error.getDescription();
+        }
+        return message;
```

Comment:
This can be simplified by returning the message inside the if condition, without assigning to a variable.

Reply:
Addressed with commit [a6f4632](https://github.com/wso2/identity-api-server/pull/315/commits/a6f463218febb07c19f260f0aa26e33fdec47de3)